llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -3062,6 +3071,26 @@ static bool isCttzOpc(unsigned Opc) {
   return Opc == ISD::CTTZ || Opc == ISD::CTTZ_ZERO_UNDEF;
 }
 
+SDValue AMDGPUTargetLowering::lowerCTLZResults(SDValue Op,
+                                               SelectionDAG &DAG) const {
+  auto SL = SDLoc(Op);
+  auto Arg = Op.getOperand(0u);
+  auto ResultVT = Op.getValueType();
+
+  if (!(ResultVT == MVT::i8 || ResultVT == MVT::i16))
```

Comment:
Nit: this can be an assert, and then this function will always succeed so the caller doesn't need a check.

Reply:
This function is also called for `ctlz.i64` since it's marked for custom lowering.